Assets/Scripts/Dialogue/InteractiveNPC.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class InteractiveNPC : MonoBehaviour
+{
+    private bool IsConversationActive;
+    private bool IsPlayerInRange;
+    private float DistanceBetweenPlayer;
+    public float MinDistance = 3.5f;
+    private DialogueManager DialogueManager;
+    public Conversation Conversation;
+    private GameObject Player;
+
+
+    void Start()
+    {
+        DialogueManager = GameObject.FindGameObjectWithTag("DialogueManager").GetComponent<DialogueManager>();
+        Player = GameObject.FindWithTag("Player");
+        IsConversationActive = false;
+        IsPlayerInRange = false;
+        DistanceBetweenPlayer = 0f;
+    }
+
+    void Update()
+    {
+        CheckIfInRange();
+
+        if (IsConversationActive && Input.GetButtonDown("Action Command"))
+        {
+            IsConversationActive = DialogueManager.AdvanceConversation();
+        }
+        else if (Input.GetButtonDown("Action Command") && IsPlayerInRange)
+        {
+            IsConversationActive = true;
+            DialogueManager.StartDialogue(Conversation);
+        }
+    }
+
+    void CheckIfInRange()
+    {
+        if (Player)
+        {
+            DistanceBetweenPlayer = Vector3.Distance(Player.transform.position, transform.position);
+        }
```

Comment:
Why are we checking for `Player`?

Reply:
do we foresee cases where the Player wont be present in a scene with only NPCs?

Reply:
I put this as an extra measure to catch the case where the player is initialized to null, should I remove the check?

Reply:
I guess it depends on whether we'd like to throw an error instead. Let's leave it for now.